tree/ntuple/v7/test/ntuple_modelext.cxx

```diff
@@ -239,6 +239,66 @@ TEST(RNTuple, ModelExtensionProject)
    EXPECT_EQ(refVec, aliasVec(1));
 }
 
+TEST(RNTuple, ModelExtensionClusterBoundaries)
+{
+   FileRaii fileGuard("test_ntuple_modelext_cluster_boundaries.root");
+   std::vector<float> refVec{1., 2., 3.};
+   CustomStruct refStruct{42.0, {1., 2., 3.}, {{4., 5.}, {}, {6}}, "foo"};
+   {
+      auto model = RNTupleModel::Create();
+
+      auto vecFld = model->MakeField<std::vector<float>>("vecFld", refVec);
+      auto structFld = model->MakeField<CustomStruct>("structFld", refStruct);
+
+      RNTupleWriteOptions opts;
+      opts.SetCompression(201);
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "myNTuple", fileGuard.GetPath(), opts);
+
+      ntuple->Fill();
```

Comment:
Not directly related to this PR: This test only one cluster, do we have a test of the late model extension (and then reading thereof) where this is more than one cluster (in particular the new column need to respect the old set of clusters).

Reply:
That's what's happening on the next lines right, or am I misunderstanding your comment?

Reply:
Ah I see what you mean! Yes, that's a good suggestion.